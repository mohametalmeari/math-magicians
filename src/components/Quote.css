.quote-container {
  display: flex;
  flex-direction: column;
  gap: 5rem;
  padding: 3rem;
  height: 100%;
  justify-content: stretch;
}

.quote-content {
  font-size: xx-large;
}

.refresh-btn {
  padding: 1rem;
}

.msg {
  padding: 1rem;
}
