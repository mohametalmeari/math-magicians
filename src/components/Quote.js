import { useEffect, useState } from 'react';

const Quote = () => {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);

  const Reload = () => {
    const fetchData = async () => {
      try {
        setIsLoading(true);
        const response = await fetch('https://api.api-ninjas.com/v1/quotes?category=happiness', {
          method: 'GET',
          headers: { 'X-Api-Key': 'oAvV8PHfcyB0hLueVYy7wQ==4Ob3080f6FPplvbt' },
          contentType: 'application/json',
        });
        const json = await response.json();
        setData(json[0].quote);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  };

  useEffect(() => {
    Reload();
  }, []);

  if (hasError) return <div className="msg">Something went wrong!</div>;

  if (isLoading) return <div className="msg">Loading...</div>;

  return (
    <div className="quote-container">
      <p className="quote-content">
        {data}
      </p>
      <button className="refresh-btn" onClick={Reload} type="button">
        Refresh
      </button>

    </div>
  );
};

export default Quote;
