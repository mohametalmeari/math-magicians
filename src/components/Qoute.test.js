import { render } from '@testing-library/react';
import Quote from './Quote';

describe('Qoute Component', () => {
  test('test render of qoute page', () => {
    const { container } = render(<Quote />);
    expect(container).toMatchSnapshot();
  });
});
