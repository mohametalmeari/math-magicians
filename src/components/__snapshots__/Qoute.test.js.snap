// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component test render of calculator page 1`] = `
<div>
  <div
    class="msg"
  >
    Loading...
  </div>
</div>
`;

exports[`Calculator Component test render of qoute page 1`] = `
<div>
  <div
    class="msg"
  >
    Loading...
  </div>
</div>
`;

exports[`Qoute Component test render of qoute page 1`] = `
<div>
  <div
    class="msg"
  >
    Loading...
  </div>
</div>
`;
